src/mlpack/methods/ann/loss_functions/poisson_nll_loss_impl.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file poisson_nll_loss_impl.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Implementation of the Poisson Negative Log Likelihood class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_POISSON_NLL_LOSS_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_POISSON_NLL_LOSS_IMPL_HPP
+
+
+// In case it hasn't yet been included.
+#include "poisson_nll_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+PoissonNLLLoss<InputDataType, OutputDataType>::PoissonNLLLoss(
+    const bool logInput,
+    const bool full,
+    const double eps,
+    const bool reduction):
+    logInput(logInput),
+    full(full),
+    eps(eps),
+    reduction(reduction)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+double PoissonNLLLoss<InputDataType, OutputDataType>::Forward(
+    const InputType& input,
+    TargetType& target)
+{
+  InputType loss;
+  loss.set_size(size(input));
```

Comment:
I don't think this line is needed as well since we are using assignment operator below. Armadillo we automatically allocate its size. Hence I feel this is redundant.

Reply:
*will

Reply:
Though I don't know how armadillo handles this but from experience with MATLAB I think initializing a vector/matrix with a predefined shape is a better practice.